app/core/ViewLoad.coffee

```diff
@@ -0,0 +1,65 @@
+VIEW_LOAD_LOG = false
+SHOW_NOTY = false
+
+class ViewLoad
+  @firstLoad: true
+  constructor: (@view) ->
+    @t0 = new Date()
+    @firstLoad = ViewLoad.firstLoad
+    ViewLoad.firstLoad = false
+
+  setView: (@view) ->
+
+  record: ->
+    console.group('Recording view:', @view.id) if VIEW_LOAD_LOG
+    views = [@view]
+    networkPromises = []
+    while views.length
+      subView = views.pop()
+      views = views.concat(_.values(subView.subviews))
+      if not subView.supermodel.finished()
+        networkPromises.push(subView.supermodel.finishLoading())
+    console.log 'Network promises:', networkPromises.length if VIEW_LOAD_LOG
+
+    Promise.all(networkPromises)
+    .then =>
+      @networkLoad = new Date()
+      return if @view.destroyed
+      
+      imagePromises = []
+      if VIEW_LOAD_LOG
+        console.groupCollapsed('Images')
+        console.groupCollapsed('Skipping')
+        for img in @view.$('img:not(:visible)')
+          console.log img.src
+        console.groupEnd()
+      for img in @view.$('img:visible')
+        if not img.complete
+          promise = new Promise((resolve) ->
+            if img.complete
+              resolve()
+            else
+              img.onload = resolve
+              img.onerror = resolve
+          )
+          promise.imgSrc = img.src
+          console.log img.src if VIEW_LOAD_LOG
+          imagePromises.push(promise)
+
+      console.groupEnd() if VIEW_LOAD_LOG
+      return Promise.all(imagePromises)
+    .then =>
+      return if @view.destroyed
+      timeStart = if @firstLoad then window.performance.timing.fetchStart else @t0
+      networkTime = @networkLoad - timeStart
+      totalTime = new Date() - timeStart
```

Comment:
Although `@networkLoad` looks like it might correspond to the timings given by `window.performance`.

For data's sake, I wonder if we should just include all three numbers — One(s) from `window.performance.timing`, ones from our own NetworkLoad (which will miss things that don't go through the supermodel), and when the images finish loading.

We might discover for example that some pages the supermodel reporting is less reliable, or that image loading takes a large fraction of the load on some pages.